Lib/test/test_minidom.py

```diff
@@ -173,6 +174,23 @@ def testAppendChild(self):
         self.assertEqual(dom.documentElement.childNodes[-1].data, "Hello")
         dom.unlink()
 
+    def testAppendChildNoQuadraticComplexity(self):
+        impl = getDOMImplementation()
+
+        newdoc = impl.createDocument(None, "some_tag", None)
+        top_element = newdoc.documentElement
+        children = [newdoc.createElement(f"child-{i}") for i in range(1, 2 ** 15 + 1)]
+        element = top_element
+
+        start = time.time()
+        for child in children:
+            element.appendChild(child)
+            element = child
+        end = time.time()
+
+        # This example used to take at least 30 seconds.
+        self.assertLess(end - start, 1)
```

Comment:
The three failures we've had:
```
AssertionError: 1.8501551151275635 not less than 1
AssertionError: 2.0126874446868896 not less than 1
AssertionError: 2.0126874446868896 not less than 1
```

All are much less than the commented "at least 30 seconds".

Shall we use something like 2.1 or 2.5 or 3 in this test?

```suggestion
        # This example used to take at least 30 seconds.
        self.assertLess(end - start, 3)
```

Reply:
3 isn't enough to make it non flaky on slow or loaded systems, or when small shifts in time happen. Frankly, this is always going to flaky, no matter the limit, because it's using time.time(). I think this is a bad idea to include in tests.

Reply:
I suggest we get this in with a check for 10 or 20 seconds just to get the release unblocked and not creating an *overly* flaky test, and continue discussion on how this could be reliably tested on main.

Reply:
```suggestion
        # This example used to take at least 30 seconds.
        self.assertLess(end - start, 10)
```

Reply:
we have several tests that assert walltime in our test suite.  my general approach has been just that: 10x the number to get the flake rate below p999 so it becomes noise and if problematic on some wide swaths of plaforms, just add skips there.  having it be executed on the steady ones where it never flakes is sufficient.